Automattic/simplenote-macos
# Interlinks Autocomplete UI: Mark IV

### Details:
- We're now filtering Interlink Autocomplete rows by **Title**
- Filtering happens in memory. Exceedingly more efficient than executing performFetch with a Blocks Based predicate.
- Improved conditions that determine when to show / hide the Interlink Autocomplete Window
- A bit of code was reorganized here and there
- And documented few methods here and there as well!

@aerych Sir!! can I bug you yeeet again?
Last one in its series!!! 🤟🤟🤟


Ref. #663

---

### Test: Filtering
1. Open any note and enable Markdown
2. Type `[keyword`

- [x] Verify that notes containing `keyword` in their Title show up
- [x] Verify that if no note contains `keyword` in their title, **nothing** shows up
- [x] Verify that selecting any of the suggestions (and pressing return) results in an Interlink onscreen

### Test: Undo!
1. Open any note and enable Markdown
2. Type `[keyword` 
- [x] Verify the Interlink Window shows up
4. Press CMD + A to highlight all the things
- [x] Verify the Interlink Window **is dismissed immediately**
6. Press CMD + X to cut
7. Press CMD + Z to undo
8. Press the right arrow
- [x] Verify the Interlink Window **does not show up**

### Test: Selection Change
1. Open any note and enable Markdown
2. Type `[keyword` 
- [x] Verify the Interlink Window shows up
3. Press **left arrow** until you're located before the first character in `keyword`
- [x] Verify the Autocomplete Window is dismissed immediately

### Release
These changes do not require release notes.
